### PR TITLE
Flight renewal form

### PR DESCRIPTION
**Pull request by davidfischer**
- Copy existing ads (you can choose them)
- Update start and end date
- Uses same targeting
- Currently the view is staff only

## Screenshot
Screenshot from 2022-02-23 11-52-29

### CODE REVIEW

File: adserver/forms.py [hunk 1]
```diff
@@ -50,6 +52,11 @@
 User = get_user_model()
 
 
+class AdvertisementMultipleChoiceField(forms.ModelMultipleChoiceField):
+    def label_from_instance(self, obj):
+        return obj
```
Comment (ericholscher) [inline adserver/forms.py:57]: This seems like an odd override class, probably worth a comment or docstring.

File: adserver/forms.py [hunk 2]
```diff
@@ -398,6 +409,129 @@ class Meta:
...
+                css_class="my-3",
+            ),
+            Submit("submit", _("Create flight via renewal")),
+        )
+        return helper
+
+    def save(self, commit=True):
+        assert commit, "Delayed saving is not supported on this form"
+
+        instance = super().save(commit=True)
```
Comment (ericholscher) [inline adserver/forms.py:496]: Should be using the `commit` variable here, even though we know it's truthy.

File: adserver/templates/adserver/includes/widgets/advertisement-form-option.html [hunk 3]
```diff
@@ -0,0 +1,50 @@
+{# Adapted from bootstrap4/layout/checkboxselectmultiple.html in crispy forms #}
```
Comment (ericholscher) [inline adserver/templates/adserver/includes/widgets/advertisement-form-option.html:1]: Not 100% sure how to review this. Might be useful to note what was changed here (eg. `Changes on lines 6-9`)

File: adserver/views.py [hunk 4]
```diff
@@ -358,6 +359,53 @@ def get_success_url(self):
...
+        return super().dispatch(request, *args, **kwargs)
+
+    def form_valid(self, form):
+        result = super().form_valid(form)
+        flight = self.object
+        messages.success(
+            self.request,
+            _("Successfully created new flight '%(flight)s' via renewal")
+            % {"flight": flight},
+        )
```
Comment (ericholscher) [inline adserver/views.py:385]: Do we want to use f-strings here?

**Review by ericholscher — APPROVED** [review]
This looks like a great enhancement! A few small nits, but excited to see this one go out.

File: adserver/templates/adserver/includes/widgets/advertisement-form-option.html (hunk 3 above)
Comment (davidfischer) [reply adserver/templates/adserver/includes/widgets/advertisement-form-option.html:1]: There are A LOT of changes. This file is now specific to advertisements.

**Pull request merged**